tests/functional/ScaffolderTest.php

```
@@ -0,0 +1,298 @@
+    {
+        $pkg = self::createStub(PackageInterface::class);
+        $pkg->method('getName')->willReturn($name);
+        $pkg->method('getExtra')->willReturn($extra);
+
+        return $pkg;
+    }
+
+    /**
+     * @param list<string> $allowedPackages
+     */
+    private function makeRootPackage(array $allowedPackages, FakeProjectBuilder $builder): PackageInterface
```

PHPMD (2.15.0)

[warning] Avoid unused parameters such as '$builder'. (undefined)

(UnusedFormalParameter)